packages/bitcore-node/src/routes/api/block.ts

```diff
@@ -53,6 +56,69 @@ router.get('/:blockId', async function(req: Request, res: Response) {
   }
 });
 
+//return all { txids, inputs, ouputs} for a blockHash paginated at max 500 per page, to limit reqs and overload
+router.get('/:blockHash/coins/:limit/:pgnum', async function (req: Request, res: Response) {
+  let { chain, network, blockHash, limit, pgnum } = req.params;
+
+  try {
+    pgnum = parseInt(pgnum, 10);
+    limit = parseInt(limit, 10);
+    if (!(typeof pgnum === "number" && typeof limit === "number")) {
+      res.status(400).send("Please enter limit and number as valid decimal numbers")
+    }
+    if (limit) {
+      if (limit > 500) limit = 500;
+    }
+  } catch (err) {
+    console.log(err);
+  }
+
+  let skips = limit * (pgnum - 1);
+  let numOfTxs: number = await TransactionStorage.collection.find({ chain, network, blockHash }).count();
+  try {
+    let txs;
+    if (numOfTxs < limit) {
+      txs = await TransactionStorage.collection.find({ chain, network, blockHash }).toArray();
```

Comment:
To get the most from TS, try to keep the declaration of variables with their assignments
```
let txs = numOfTxs < limit ?
    find() :
    find().skip().limit()
```
or

```
const query = TransactionStorage.collection.find({ chain, network, blockHash });
const txs = await numOfTxs < limit ? query.toArray() : query.skip().limit()
```